### PR TITLE
Provide 24.10 version bumps

### LINKED ISSUE #403: [FEA] Support XGBoost UBJSON model format
cuML 24.08 added support for XGBoost UBJSON.

### PR DESCRIPTION
**Pull request by hcho3**
* Add support for XGBoost UBJSON
* Bump package versions in Conda envs
* Replace `convert_sklearn` script to use latest Treelite API
* Update `generate_example_model.py` to use latest XGBoost API

Closes #403

### CODE REVIEW

**Review by wphicks — DISMISSED** [review]

File: qa/entrypoint.sh [hunk 1]
```diff
@@ -22,6 +22,7 @@ if [[ $TRITON_ENABLE_GPU != "OFF" ]]
 then
   echo 'Running tests for GPU models...'
   MODEL_REPO="${QA_DIR}/L0_e2e/model_repository" "$TEST_SCRIPT"
+  sleep 3
```
Comment (wphicks) [inline qa/entrypoint.sh:25]: Could you add a comment detailing exactly why this is necessary?

Comment (hcho3) [reply qa/entrypoint.sh]: This change is no longer in effect.

**Comment by hcho3** [comment]
The CI is now all green.

**Comment by hcho3** [comment]
@wphicks Hi, I updated the tests to XFAIL when the shared mem is used.

File: qa/L0_e2e/test_model.py [hunk 2]
```diff
@@ -66,6 +67,16 @@ def valid_shm_modes():
     return tuple(modes)
 
 
+# TODO(hcho3): Remove once we fix the flakiness of CUDA shared mem
```
Comment (wphicks) [inline qa/L0_e2e/test_model.py:70]: Could we link explicitly to the corresponding Triton Github issue?

File: qa/L0_e2e/test_model.py [hunk 3]
```diff
@@ -98,27 +109,46 @@ class GTILModel:
...
-        else:
-            self.tl_model = treelite.Model.load(model_path, model_format)
         self.output_class = output_class
 
     def _predict(self, arr):
-        return treelite.gtil.predict(self.tl_model, arr)
+        result = treelite.gtil.predict(self.tl_model, arr)
+        # GTIL always returns prediction result with dimensions
+        # (num_row, num_target, num_class)
+        assert len(result.shape) == 3
```
Comment (wphicks) [inline qa/L0_e2e/test_model.py:134]: Why are we asserting anything in this method? This method is used just to get a "golden" value for comparison with results from Triton, right?

File: qa/run_tests.sh [hunk 4]
```diff
@@ -102,9 +102,10 @@ trap finally EXIT
 if [ ! -z $CPU_ONLY ] && [ $CPU_ONLY -eq 1 ]
 then
   pytest \
+    -v -rxXs \
```
Comment (wphicks) [inline qa/run_tests.sh:105]: Do we always want `-v` in CI runs?

File: scripts/convert_cuml.py [hunk 5]
```diff
@@ -21,18 +21,18 @@
 """
 
 import argparse
-import os
+import pathlib
```
Comment (wphicks) [inline scripts/convert_cuml.py:24]: In general, let's try to leave unrelated changes out of PRs, especially version bumps PRs. Ideally, this would be the minimum changeset we need to bump to latest versions. No need to change this back, though; it's reasonably innocuous.

File: qa/run_tests.sh (hunk 4 above)
Comment (hcho3) [reply qa/run_tests.sh:105]: My bad, I added it for local testing and accidentally committed it. Let me remove it.

File: qa/L0_e2e/test_model.py (hunk 3 above)
Comment (hcho3) [reply qa/L0_e2e/test_model.py:134]: In the future where we add multi-target models, the tests should fail so that we can revise them. In general, I like to use asserts to explicitly state assumptions being made about a test (here, we assume `n_outputs == 1`).

File: qa/L0_e2e/test_model.py (hunk 2 above)
Comment (hcho3) [reply qa/L0_e2e/test_model.py:70]: It took me a while to craft a minimal reproducer: https://github.com/triton-inference-server/server/issues/7688. Let me link it now.
